bucoffea/vbfhinv/definitions.py

```diff
@@ -326,188 +333,94 @@ def vbfhinv_regions(cfg):
+    to_remove = ['veto_ele']
+    regions['cr_1e_vbf'] = clean_lists( add_lists(to_add, common_cuts), to_remove)
+
+    # Photon CR
+    to_add = ['trig_photon', 'at_least_one_tight_photon', 'one_photon', 'photon_pt', 'dpfcalo_cr']
+    to_remove = ['veto_photon']
+    regions['cr_g_vbf'] = clean_lists( add_lists(to_add, common_cuts), to_remove)
 
     # Z CRs with CaloMETNoLep cut
     if cfg.RUN.CALOMET_CHECK:
         for r in ['cr_2e_vbf', 'cr_2m_vbf']:
-            regions[f'{r}_calocut'] = copy.deepcopy(regions[r])
-            regions[f'{r}_calocut'].append('calo_metptnolep')
-
-    # Single electron CR
-    cr_1e_cuts = ['trig_ele','one_electron', 'at_least_one_tight_el', 'veto_muo','met_el'] + common_cuts[1:] + ['dpfcalo_cr', 'no_el_in_hem']
-    # cr_1e_cuts.remove('veto_ele')
-    regions['cr_1e_vbf'] =  cr_1e_cuts
-
-    # Photon CR
-    cr_g_cuts = ['trig_photon', 'one_photon', 'at_least_one_tight_photon','photon_pt'] + common_cuts + ['dpfcalo_cr']
-    cr_g_cuts.remove('veto_photon')
-
-    regions['cr_g_vbf'] = cr_g_cuts
-
-    if cfg and cfg.RUN.SYNC:
-        regions['sync_sr_vbf_round1'] = [
-                                        'filt_met',
-                                        'trig_met',
-                                        'veto_photon',
-                                        'mindphijr',
-                                        'recoil',
-                                        'two_jets',
-                                        'leadak4_pt_eta',
-                                        'leadak4_id',
-                                        'trailak4_pt_eta',
-                                        'trailak4_id',
-                                        'hemisphere',
-                                        'mjj',
-                                        'dphijj',
-                                        'detajj',
-                                        ]
+            regions[f'{r}_calocut'] = add_lists(regions[r], 'calo_metptnolep')
 
     # VBF signal region where the hard-lepton vetoes are replace
     # with lepton veto weights
-    tmp = {}
-    for region in regions.keys():
-        if not region.startswith("sr_"):
-            continue
-        new_region = f"{region}_no_veto_all"
-        tmp[new_region] = copy.deepcopy(regions[region])
-        tmp[new_region].remove("veto_muo")
-        tmp[new_region].remove("veto_tau")
-        tmp[new_region].remove("veto_ele")
-        tmp[new_region].remove("mindphijr")
-        tmp[new_region].remove("recoil")
-        tmp[new_region].append("met_sr")
-        tmp[new_region].append("mindphijm")
-
-    regions.update(tmp)
+    to_add = ['met_sr', 'mindphijm']
+    to_remove = ['veto_muo', 'veto_tau', 'veto_ele', 'mindphijr', 'recoil']
+    regions.update(dict([(f"{region}_no_veto_all", add_lists(clean_lists(regions[region], to_remove),to_add)) for region in regions.keys() if region.startswith("sr_")]))
```

Comment:
I love the one-liner, but it might be confusing for new people when looking at the code to understand what this does I think :) What do you think? Can you maybe at least make the `for` loop explicit like the previous version? Or if you want to keep it this way, maybe a comment on top explaining what this does might be helpful.

Reply:
Done. Let me know if you want some more clarity more